yrs/src/update.rs

```diff
@@ -247,32 +249,35 @@ impl Update {
                     stack.push(block);
                     // hid a dead wall, add all items from stack to restSS
                     Self::return_stack(stack, &mut self.blocks, &mut remaining);
-                    current_target =
-                        current_client_id.and_then(|id| self.blocks.clients.get_mut(&id));
+                    current_target = self.blocks.clients.get_mut(&current_client_id);
                     stack = Vec::new();
                 }
 
                 // iterate to next stackHead
                 if !stack.is_empty() {
                     stack_head = stack.pop();
                 } else {
-                    current_target = match current_target.take() {
-                        Some(v) if !v.is_empty() => Some(v),
+                    match current_target.take() {
+                        Some(v) if !v.is_empty() => {
+                            stack_head = v.pop_front();
```

Comment:
There was some error here as well, but due to magic of Git cherry-pick I can no longer recall what was wrong.